include/seqan3/core/platform.hpp

```diff
@@ -97,7 +97,23 @@
 #endif
 
 // SDSL [required]
-// TODO (doesn't have a version.hpp, yet)
+#if __has_include(<sdsl/version.hpp>)
+#   define SDSL_MINVERSION 30000
+#   define SDSL_MAXVERSION 39999
+#   define MSG "Your version: " STR(SDSL_VERSION) \
+                "; minimum version: " STR(SDSL_MINVERSION) \
+                "; expected maximum version: " STR(SDSL_MAXVERSION)
+#   include <sdsl/version.hpp>
+#   if SDSL_VERSION < SDSL_MINVERSION
+#       error Your sdsl library is too old.
+#       pragma message(MSG)
+#   elif SDSL_VERSION > SDSL_MAXVERSION
+#       pragma GCC warning "Your sdsl library is possibly too new. Some features might not work correctly."
+#       pragma message(MSG)
+#   endif
```

Comment:
Since you have added static constexpr version info to the sdsl-file, you can also use C++ to enforce the version, simply do this instead of the macro stuff:
```cpp
static_assert(sds_version_major == 3, "Only version 3 of the SDSL is supported by SeqAn3.");
```

:smile:

Reply:
But this would trigger an error for sdsl 4.x (instead of a warning).